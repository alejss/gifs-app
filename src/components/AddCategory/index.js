import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ seetCategories }) => {
    const [inputValue, setinputValue] = useState('')

    const handleChangeText = (evn) => {
        const text = evn.target.value
        setinputValue(text)
    }

    const handleSubmit = (evn) => {
        evn.preventDefault()
        if (inputValue.trim().length > 2) seetCategories(categories => [inputValue, ...categories])
        setinputValue('')
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type='text'
                placeholder='Add categories...'
                value={inputValue}
                onChange={handleChangeText}
            />
        </form>
    )
}

AddCategory.propTypes = {
    seetCategories: PropTypes.func.isRequired,
}
