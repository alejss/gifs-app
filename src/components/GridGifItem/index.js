import React from 'react'
import './style.css'

export const GridGifItem = ({ id, title, url }) => {
    return (
        <div className="card animate__animated animate__swing">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
