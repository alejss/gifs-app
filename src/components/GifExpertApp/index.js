import React, { useState } from 'react'
import './style.css'

// Components
import { AddCategory } from '../AddCategory'

export const GifExpertApp = () => {
  const [categories, seetCategories] = useState(['Cats', 'Dogs', 'Random'])
  return (
    <>
      <h2>Gif Expert App</h2>
      <hr />
      <AddCategory seetCategories={seetCategories} />
      <ol>
        {
          categories.map((categorie) => (<li key={categorie}>{categorie}</li>))
        }
      </ol>
    </>
  )
}
