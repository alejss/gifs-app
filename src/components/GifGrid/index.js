import React, { useState, useEffect } from 'react'
import './style.css'
import { GridGifItem } from '../GridGifItem'

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs()
    }, [])

    const getGifs = async () => {
        const url = `https://api.giphy.com/v1/gifs/search?q=Cats&limit=10&api_key=${process.env.REACT_APP_GIPHY_KEY}`
        const resp = await fetch(url)
        const { data } = await resp.json()

        const gifs = data.map(({ id, title, images }) => ({ id, title, url: images.downsized_medium.url }))
        setImages(gifs)
    }

    getGifs()

    return (
        <>
            <h3 key={category}>{category}</h3>
            <div className="card-grid">
                {
                    images.map((image) => (
                        <GridGifItem
                            key={image.id}
                            {...image}
                        />)
                    )
                }
            </div>
        </>
    )
}
