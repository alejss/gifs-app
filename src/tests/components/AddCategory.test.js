import { shallow } from 'enzyme';
import React from 'react';
import { AddCategory } from '../../components/AddCategory';

describe('Testing <AddCategory/>', () => {

    const setCategories = jest.fn()
    let wrapper

    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory  {...{ setCategories }} />)
    })

    test('Testing Snapshot Component', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Input box text changed', () => {
        const input = wrapper.find('input')
        const value = 'Hi everyone'
        input.simulate('change', { target: { value } })

        expect(wrapper.find('p').text().trim()).toBe(value)
    })

    test('You should not submit the information', () => {
        wrapper.find('form').simulate('submit', { preventDefault: () => { } })
        expect(setCategories).not.toHaveBeenCalled()
    })

    test('Run setCategories and clean the text', () => {
        const value = 'Goku';
        wrapper.find('input').simulate('change', { target: { value }});
        wrapper.find('form').simulate('submit', { preventDefault: () => { } })
 
        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
    })



});