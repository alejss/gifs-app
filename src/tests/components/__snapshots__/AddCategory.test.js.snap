// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AddCategory/> Testing Snapshot Component 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    placeholder="Add categories..."
    type="text"
    value=""
  />
</form>
`;
